modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/K6ClientCodegen.java

```diff
@@ -744,9 +885,275 @@ private static String getAccept(OpenAPI openAPI, Operation operation) {
+        }
+
+        if (operationId.isPresent() && valuePath.isPresent() && parameterName.isPresent()) {
+            return Optional
+                    .of(new DataExtractSubstituteParameter(operationId.get(), valuePath.get(), parameterName.get()));
+
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Optionally, retrieve data extraction properties for given operation
+     *
+     * @param dataExtractSubstituteParams
+     * @param operationId
+     * @return optional as only returned if present for given operation
+     */
+    private Optional<DataExtractSubstituteParameter> getDataExtractSubstituteParameter(
+            Map<String, DataExtractSubstituteParameter> dataExtractSubstituteParams, String operationId) {
+
+        return (!dataExtractSubstituteParams.isEmpty() && dataExtractSubstituteParams.containsKey(operationId))
+                ? Optional.of(dataExtractSubstituteParams.get(operationId))
+                : Optional.empty();
+    }
+
+    /**
+     * Optionally, retrieve information specified in the `X_OPERATION_GROUPING` K6
+     * vendor extension
+     *
+     * @param cgOperation
+     * @return optional as only returned if required info is present
+     */
+    private Optional<OperationGrouping> extractOperationGrouping(CodegenOperation cgOperation) {
+        Optional<OperationGrouping> operationGrouping = Optional.empty();
+
+        if (cgOperation.getHasVendorExtensions() && cgOperation.vendorExtensions.containsKey(X_OPERATION_GROUPING)
+                && cgOperation.vendorExtensions.get(X_OPERATION_GROUPING) instanceof java.util.Map) {
+
+            Map.Entry<?, ?> operationGroupingEntry = ((Map<?, ?>) cgOperation.vendorExtensions
+                    .get(X_OPERATION_GROUPING)).entrySet().stream().findFirst().get();
+
+            return Optional.of(new OperationGrouping(String.valueOf(operationGroupingEntry.getKey()),
+                    Integer.parseInt(String.valueOf(operationGroupingEntry.getValue()))));
+        }
+
+        return operationGrouping;
+    }
+
+    /**
+     * If `X_OPERATION_RESPONSE` K6 vendor extension is present, check if given
+     * operation response should be hidden.
+     *
+     * @param resp
+     * @return true if should be hidden, false otherwise
+     */
+    private boolean shouldHideOperationResponse(ApiResponse resp) {
+        boolean hideOperationResponse = false;
+
+        if (Objects.nonNull(resp.getExtensions()) && !resp.getExtensions().isEmpty()
+                && resp.getExtensions().containsKey(X_OPERATION_RESPONSE)) {
+
+            Map<?, ?> respExtensions = (Map<?, ?>) resp.getExtensions().get(X_OPERATION_RESPONSE);
+
+            for (Map.Entry<?, ?> respExtensionEntry : respExtensions.entrySet()) {
+
+                if (X_OPERATION_RESPONSE_HIDE.equals(String.valueOf(respExtensionEntry.getKey()))) {
+                    hideOperationResponse = Boolean.parseBoolean(respExtensionEntry.getValue().toString());
```

Comment:
should this have a `break` statement after setting this to true? effectively, now `hideOperationResponse` will have the last value of `respExtensionEntry.getValue()`, and if you want the last value, then we don't need a loop here, and since we're iterating on `respExtensions.entrySet()` a Set, the last value is not deterministic, so method might produce different value on every run.

Reply:
Addressed.